### PR TITLE
Tests for PluginMap Plugin

### LINKED ISSUE #141: Phase 5 : Implement Comprehensive Test Suite for Plugin Map
**Is your feature request related to a problem? Please describe.**
The `Plugin Map` plugin, while a critical developer tool for visualizing extension points in the Talawa Admin system, currently lacks a comprehensive automated test suite. This makes it difficult to ensure that the plugin remains functional as the core platform evolves, and risks regressions in both the data-fetching layer (API) and the visual indicators (Admin).

**Describe the solution you'd like**
I would like to implement a robust test suite for the `Plugin Map` plugin, targeting approximately 90% statement and branch coverage. This involves:
1.  **Backend (API) Tests**: Covering GraphQL resolvers (queries/mutations) and plugin lifecycle hooks
2.  **Frontend (Admin) Tests**: Covering the main Explorer pages, Dashboard, and the `MapIconInjector` to verify that UI components are correctly placed and interactive.

**Approach to be followed (optional)**
1.  **Utility Setup**: Create mock utilities (e.g., `createMockPluginMapContext`) following the patterns established for the Razorpay plugin.
2.  **API Layer Verification**:
    *   Test `queries.ts` to ensure it correctly returns system-wide extension points.
    *   Test lifecycle hooks to verify proper registration in the system database.
3.  **Admin Layer Verification**:
    *   Implement component tests for `ExtensionPointsDashboard`, `ExtensionPointsUser`, etc., verifying permission-gate logic (Admin vs. User views).
    *   Test `MapIconInjector` to ensure markers correctly appear when extension points are available.
4.  **Coverage Goal**: Aim for ~70-80 individual test cases to account for edge cases like missing organizations, permission denials, and empty system states.

**Additional context**
The current `Plugin Map` codebase consists of approximately 2,500 lines across API and Admin directories. Achieving >=90% coverage will require a systematic approach to mocking the Talawa plugin host environment and the Drizzle ORM layer. This could take up to 3-4 working days.

### PR DESCRIPTION
**Pull request by Ansingh0305**
**What kind of change does this PR introduce?**

* Testing: Finalization of Plugin Map test suite.
* Refactoring: Resolution of project-wide linting and TypeScript-checking errors.
* Documentation: Restoration of JSDoc examples in shared types.

**Issue Number:**

Fixes #141 

**Snapshots/Videos:**

NA

**If relevant, did you update the documentation?**

Yes, regenerated and fixed JSDoc documentation for the Plugin Map and Razorpay plugins.

**Summary**

This PR completes the testing initiative for the Plugin Map plugin and addresses technical debt across the codebase.

**Tests Added & Coverage Results:**
* **Admin Layer Tests**: Dashboard, Global View, Organization View, and User View components (Achieved **100% coverage**).
* **API Layer Tests**: Plugin Lifecycle Hooks, GraphQL Queries, and GraphQL Mutations (Achieved **100% coverage**).
* **System Layer Tests**: Plugin registration and automated database table verification.

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI.
- [ ] I have implemented or provided justification for non-critical suggestions.

### Test Coverage
- [x] I have written tests for all new changes/features.
- [x] I have verified that test coverage meets 100% for Plugin Map.
- [x] I have run the test suite locally and all tests pass.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-plugin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **New Features**
  * Full i18n for Plugin Map admin UI (English resource, init) and ARIA labels; plugin context can optionally expose GraphQL and Pub/Sub.

* **Bug Fixes**
  * More reliable data fetching (network-only, hook ordering) and unified translated success/error messaging; structured error logging across lifecycle and payment flows.

* **Chores**
  * Centralized GraphQL operations, stronger typings, CI workflow tweaks, and updated docs.

* **Tests**
  * Extensive unit and UI test suites, shared test utilities, and Ant Design test mocks.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: plugins/Plugin Map/admin/graphql/queries.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { gql } from '@apollo/client';
+
+export const GET_PLUGIN_MAP_REQUESTS = gql`
+  query GetPluginMapRequests($input: GetPluginMapRequestsInput) {
+    plugin_map_getPluginMapRequests(input: $input) {
+      requests {
```
Comment (palisadoes) [inline plugins/Plugin Map/admin/graphql/queries.ts]: Why is the parent directory not `/src/pluginMap` ?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See basic questions

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:

1. **Test Coverage & Quality Validation:**
   1. Validate whether any updated test files:
      1. Cover all use cases including edge cases, error scenarios, and success paths
      2. Have valid, properly structured tests following Jest and React Testing Library best practices
      3. Provide 100% test code coverage for all modified/added functions, components, and branches
      4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
      5. Test both user interactions and state changes comprehensively
      6. Require any additional tests for uncovered scenarios
   2. For each modified file with logic or UI, verify corresponding test files exist and are updated
   3. Check that test files follow naming conventions (*.test.tsx, *.test.ts, *.spec.tsx, *.spec.ts)
   4. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
   5. Ensure mock data cleanup is performed (as required by pre-commit hooks)

2. **Code Quality & Repository Standards:**
   1. Verify TypeScript compliance:
      1. No use of `any` type without explicit justification
      2. Proper interface/type definitions for all props, state, and function parameters
      3. No TypeScript errors or warnings
   2. Validate GraphQL implementation:
      1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
      2. No duplicate or redundant queries (check for pagination inconsistencies)
      3. Error handling is implemented for all GraphQL operations
   3. Check React component best practices:
      1. Functional components with proper hooks usage
      2. Material-UI component patterns are followed consistently
      3. No prop drilling; proper use of context when needed
      4. Proper cleanup in useEffect hooks
   4. Verify internationalization (i18n):
      1. All user-facing strings use i18next translation keys
      2. New translation keys are added to locale files
      3. No hardcoded English text in UI components
   5. Validate accessibility:
      1. Proper ARIA labels and roles
      2. Keyboard navigation support
      3. Screen reader compatibility

3. **Documentation & Code Comments:**
   1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
   2. Check for adequate JSDoc comments on public functions and complex logic
   3. Ensure README.md or related docs are updated if new features are added
   4. Validate that Table of Contents is updated (as required by pre-commit hooks)

4. **Identify Issues:**
   1. Lines of code that are missing tests (list specific line numbers)
   2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
   3. Unused imports or dead code (as checked by knip)
   4. Console.log statements or debug code left in production files
   5. Files that should be excluded from the PR scope

5. **PR Requirements Validation:**
   1. Verify the PR meets the requirements of the linked issue:
      1. All acceptance criteria are addressed
      2. No out-of-scope changes are included
   2. Validate compliance with all previously suggested improvements and comments:
      1. All review comments have been addressed
      2. Requested changes have been implemented correctly
   3. Check PR metadata:
      1. Proper title and description
      2. Linked to appropriate issue(s)
      3. Target branch is correct (typically `develop`)
   4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)

6. **CI/CD Pipeline Compatibility:**
   1. Validate that code changes won't break any of the 17+ CI jobs
   2. Ensure pre-commit hook requirements are met:
      1. Code formatting (Prettier)
      2. Linting (ESLint)
      3. Type checking (TypeScript)
   3. Check that changes don't introduce performance regressions for the 12-shard test execution

7. **Security & Best Practices:**
   1. No security vulnerabilities introduced
   2. No exposed API keys, tokens, or credentials
   3. Proper input validation and sanitization
   4. No SQL injection or XSS vulnerabilities in any data handling

8. **Review Decision:**
    1. When critical issues are found, clearly state that changes are required in your response but DO NOT append  the APPROVED tag.
    1.  Only append  the APPROVED tag when ALL criteria are satisfied and zero changes are required:
        1. Complete test coverage (≥95%)
        1. No code quality issues, security concerns, or bugs
        1. Full implementation of issue requirements
        1. Compliance with all previous feedback
        1. No unnecessary files

9. **Focus Areas:**
   - Prioritize substantive issues over minor stylistic suggestions
   - Focus on issues that would cause CI pipeline failures or production bugs
   - Highlight any deviations from established codebase patterns

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -348,8 +344,8 @@ jobs:
           # Get the base branch ref
           BASE_SHA=$(git merge-base ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }})
 
-          # Get all changed files
-          ALL_CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRT $BASE_SHA ${{ github.event.pull_request.head.sha }} | tr '\n' ' ')
+          # Get all changed files with quoting
+          ALL_CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRT $BASE_SHA ${{ github.event.pull_request.head.sha }} | sed 's/^/"/; s/$/"/' | tr '\n' ' ')
```
[the remark this answers is not in the document]
Comment (Ansingh0305) [reply .github/workflows/pull-request.yml]: The quoting is necessary to handle file paths that contain spaces, specifically for the Plugin Map directory.
Without the quotes, the CI scripts interpret the space in plugins/Plugin Map as a separator, checking for two non-existent files (plugins/Plugin and Map) instead of one. This caused the validation scripts to fail. Quoting the paths ensures they are treated as a single valid file path.

**Comment by Ansingh0305** [comment]
@coderabbitai I have added the jsdoc commits, the useEffect cleanup is not required since the hooks are simple (triggering refetches or state updates) and don't create subscriptions, timers, or event listeners that need explicit teardown.

File: plugins/Plugin Map/admin/graphql/queries.ts (hunk 1 above)
Comment (palisadoes) [reply plugins/Plugin Map/admin/graphql/queries.ts]: Sorry, I meant /src/plugins/pluginMap similar to what you have in the test directory structure

**Comment by Ansingh0305** [comment]
### Regarding `/src/plugins/` Structure

I noticed the mention of `/src/plugins/pluginMap`. After researching best practices, I found that a dedicated `src` folder is typically used for *main application source code* (controllers, services, models, etc.). However, since this repository (`talawa-plugin`) is specifically a standalone plugin development repository—where each plugin in `plugins/` is essentially its own self-contained module with `api/`, `admin/`, and `app/` subfolders—keeping `plugins/` at the root level is appropriate and src is not necessary. This approach is consistent with how the other existing plugins (`Razorpay`, `SummarizeAPI`, `SummarizeT5`) are currently structured. If you'd prefer a different structure, please let me know.

**Comment by palisadoes** [comment]
> ### Regarding `/src/plugins/` Structure
> 
> I noticed the mention of `/src/plugins/pluginMap`. After researching best practices, I found that a dedicated `src` folder is typically used for _main application source code_ (controllers, services, models, etc.). However, since this repository (`talawa-plugin`) is specifically a standalone plugin development repository—where each plugin in `plugins/` is essentially its own self-contained module with `api/`, `admin/`, and `app/` subfolders—keeping `plugins/` at the root level is appropriate and src is not necessary. This approach is consistent with how the other existing plugins (`Razorpay`, `SummarizeAPI`, `SummarizeT5`) are currently structured. If you'd prefer a different structure, please let me know.

I was asking about the fact that this directory path has a space in it which won't lint without special exceptions. We should be using standard naming conventions.

```
plugins/Plugin Map/admin/graphql/queries.ts
```
